lib/autoupdate/agent/process.go

```diff
@@ -109,27 +117,67 @@ func (s SystemdService) Reload(ctx context.Context) error {
 	default:
 		s.Log.InfoContext(ctx, "Gracefully reloaded.", unitKey, s.ServiceName)
 	}
+	// monitor logs all relevant errors, so we filter for a few outcomes
+	err = s.monitor(ctx, initPID)
+	if errors.Is(err, context.DeadlineExceeded) {
+		return trace.Errorf("timed out while waiting for process to start")
+	}
+	if errors.Is(err, context.Canceled) {
+		return context.Canceled
+	}
+	if err != nil {
+		return trace.Errorf("failed to monitor process")
+	}
+	return nil
+}
+
+// monitor for a started, healthy process.
+// monitor logs all errors that should be displayed to the user.
+func (s SystemdService) monitor(ctx context.Context, initPID int) error {
+	ctx, cancel := context.WithTimeout(ctx, monitorTimeout)
+	defer cancel()
+	tickC := time.NewTicker(monitorInterval).C
```

Comment:
Is the tick channel shared to deliberately do at most one thing every few seconds? If so it deserves a comment to point that out.

Reply:
No particular reason to use the same clock for both checks. Re-initialized it.